### ISSUE TITLE
BUG: `prefer-native-coercion-functions` false positive in type guards

### ISSUE THREAD

**Issue by ShayanTheNerd** [sindresorhus/eslint-plugin-unicorn#3140]
ESLint version: 10.4.1
eslint-plugin-unicorn version: 65.0.1

_eslint.config.ts_:
```ts title="eslint.config.ts"
import { defineConfig } from 'eslint/config';
import eslintPluginUnicorn from 'eslint-plugin-unicorn';

export default defineConfig([{
  name: 'project/unicorn',
  files: ['**/*.?([mc])[jt]s?(x)'],
  plugins: {
    unicorn: eslintPluginUnicorn,
  },
  rules: {
    'no-implicit-coercion': 'error',
    'unicorn/prefer-native-coercion-functions': 'warn',
  },
}]);
```

_index.ts_:
```ts title="index.ts"
// Similar to the last example from the docs:
// https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-native-coercion-functions.md#examples
const mixedData = ['string', null, undefined];
const strings = mixedData.filter((value): value is string => Boolean(value));
```
✅ Expected behavior: no linting issues.
❌ Actual behavior (the false positive):
> arrow function is equivalent to `Boolean`. Use `Boolean` directly. — eslint(`unicorn/prefer-native-coercion-functions`)
